src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -1273,7 +1273,7 @@ public SchemaShape GetOutputSchema(SchemaShape inputSchema)
                     metadata.Add(slotMeta);
                 if (colInfo.InvertHash != 0)
                     metadata.Add(new SchemaShape.Column(AnnotationUtils.Kinds.KeyValues, SchemaShape.Column.VectorKind.Vector, TextDataViewType.Instance, false));
-                result[colInfo.Name] = new SchemaShape.Column(colInfo.Name, col.ItemType is VectorType ? SchemaShape.Column.VectorKind.Vector : SchemaShape.Column.VectorKind.Scalar, NumberDataViewType.UInt32, true, new SchemaShape(metadata));
+                result[colInfo.Name] = new SchemaShape.Column(colInfo.Name, col.Kind, NumberDataViewType.UInt32, true, new SchemaShape(metadata));
             }
```

Comment:
Does it mean input shape must be scalar? If yes, we need to throw when encountering a vector.

Reply:
No. It means we accept vectors, scalars and varvectors

---
In reply to: [263166153](https://github.com/dotnet/machinelearning/pull/2681#discussion_r263166153) [](ancestors = 263166153)